### PR TITLE
Update: Improved compilation times (#3358)

### LINKED ISSUE #3358: Compilation improvements
### Subject of the issue
Switch to [terser](https://www.npmjs.com/package/@rollup/plugin-terser/v/0.3.0) for minification as can be substantially faster.

### PR DESCRIPTION
**Pull request by oliverfoster**
fixes #3358

### Fix
* Tightened the glob statements for less, templates, libraries, fonts, assets and required folders
  * !! All relevent files must live in their respective folders at the root of the plugin 
* Updated compilation modules in package.json
* Switched out grunt-contrib-uglify and babeloutputplugin for terser, such that bundled modules are minified faster and libraries files minification happens faster
* Decreased the amount of compression in the compilation cache file in favour of speed
* Updated vars to consts in changed files

### Before
image

### After
image

### CODE REVIEW

File: grunt/config/less.js [hunk 1]
```diff
@@ -89,10 +90,10 @@ module.exports = function(grunt, options) {
           '<%= sourcedir %>core/less/**/*.less'
         ],
         src: [
-          '<%= sourcedir %>components/**/*.less',
-          '<%= sourcedir %>extensions/**/*.less',
-          '<%= sourcedir %>menu/<%= menu %>/**/*.less',
-          '<%= sourcedir %>theme/<%= theme %>/**/*.less'
+          '<%= sourcedir %>components/*/less/**/*.less',
+          '<%= sourcedir %>extensions/*/less**/*.less',
```
Comment (cahirodoherty-learningpool) [inline grunt/config/less.js:94]: Missing backslash after 'less' here?

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Comment by taylortom** [comment]
:tada: I'll give this a spin in the new AAT v1

**Comment by oliverfoster** [comment]
@taylortom any luck?

**Comment by taylortom** [comment]
@oliverfoster just tried this on the super slow testing server, and is much much quicker :smile::+1:

**Review by taylortom — APPROVED** [review]

**Pull request merged**